lib/child_process.js

```diff
@@ -712,12 +712,12 @@ function normalizeSpawnArguments(file, args, options) {
   };
 }
 
-function abortChildProcess(child, killSignal) {
+function abortChildProcess(child, killSignal, reason) {
   if (!child)
     return;
   try {
     if (child.kill(killSignal)) {
-      child.emit('error', new AbortError());
+      child.emit('error', new AbortError(undefined, { cause: reason }));
```

Comment:
We should not set the cause property when there are no reason I think

Reply:
If the cause is not given we would default to DOMException no? That should cover it